src/wallet/walletdb.cpp

```diff
@@ -637,6 +637,9 @@ bool ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue, CW
             ssValue >> pSettings;
             pwallet->fCombineDust = pSettings.first;
             pwallet->nAutoCombineThreshold = pSettings.second;
+            // originally saved as integer
+            if (pwallet->nAutoCombineThreshold < COIN)
+                pwallet->nAutoCombineThreshold *= COIN;
```

Comment:
this will create issues if someone sets a threshold below 1 coin.
Maybe we should just use a different key in the database, e.g. `autocombinesettingsV2`.

Reply:
It's currently (`master` branch) not possible to set the threshold below `1` as the `autocombine` RPC command only accepts whole numbers. So this conversion is "safe" in the sense of upgrading from an older wallet that had an int value set previously.

I think it would be acceptable to make it a local policy in the new command to also not allow values less than `1` as such a low threshold is basically "dust" and transaction fees would make it pretty uneconomic/unreasonable. Even further, the biggest use case for autocombine is to combine multiple 3 PIV masternode rewards into a larger single UTXO, so setting a threshold less than `3` wouldn't be effective in achieving this.

Reply:
Yes. My only concern was about *after* this change, when it could be possible to set a threshold of, say, `0.99`.
I'd be ok with introducing a `1 PIV` limit in the RPC.